src/sinks/datadog/metrics/encoder.rs

```diff
@@ -735,9 +735,8 @@ fn source_type_to_service(source_type: &str) -> Option<u32> {
         // Generally that means the Origin Metadata will have been set as a pass through.
         // However, if the upstream Vector instance did not set Origin Metadata (for example if it is an
         // older version version), we will at least set the OriginProduct and OriginCategory.
-        "kafka" | "nats" | "redis" | "gcp_pubsub" | "http_client" | "http_server" | "vector" => {
-            Some(0)
-        }
+        "kafka" | "nats" | "redis" | "gcp_pubsub" | "http_client" | "http_server" | "vector"
+        | "pulsar" => Some(0),
```

Comment:
Is the above comment, `These sources are only capable of receiving metrics with the native or native_json codec.`, true for pulsar or these other sources?

Reply:
Good question. @neuronull do you know if this is the case?

Reply:
Looking at the outputs from the `SourceConfig` in this PR, it does appear that this new source is one where if it receives metrics, its only possible from the native codecs, due to `self.decoding.output_type(),`.